main.tf

```diff
@@ -14,7 +14,7 @@ resource "aws_sns_topic_subscription" "this" {
 
 resource "aws_budgets_budget" "this" {
   count             = var.enabled ? 1 : 0
-  name              = "budget-${var.cost_filters_service}-${var.env}-${var.time_unit}"
+  name              = "budget-${var.cost_filters_service}-${var.env}-${provider.aws.region}-${var.time_unit}"
```

Comment:
could potentially see running this across multiple environments to check for loose ends